nautobot/docs/development/getting-started.md

```diff
@@ -22,52 +36,158 @@ LICENSE.txt      development  mkdocs.yml  pyproject.toml           upgrade.sh
+
+- `origin` - The default remote name used to refer to *your fork of Nautobot* 
+- `upstream` - The main remote used to refer to the *official Nautobot repository*
+
+### Setting up your Remotes
+
+Remote repos are managed using the `git remote` command. 
+
+Upon cloning Nautobot for the first time, you will have only a single remote:
+
+```no-highlight
+$ git remote -v
+origin	git@github.com:yourusername/nautobot.git (fetch)
+origin	git@github.com:yourusername/nautobot.git (push)
+```
+
+Add the official Nautobot repo as a the `upstream` remote:
+
+```no-highlight
+$ git remote add upstream git@github.com:nautobot/nautobot.git
+```
+
+View your remotes again to confirm you've got both `origin` pointing to your fork and `upstream` pointing to the official repo:
+
+```no-highlight
+$ git remote -v
+origin	git@github.com:yourusername/nautobot.git (fetch)
+origin	git@github.com:yourusername/nautobot.git (push)
+upstream	git@github.com:nautobot/nautobot.git (fetch)
+upstream	git@github.com:nautobot/nautobot.git (push)
+```
+
+You're now ready to proceed to the next steps.
+
+!!! hint
+	You will always **push** changes to `origin` (your fork) and **pull** changes from `upstream` (official repo).
+
+### Creating a Branch
+
+Before you make any changes, always create a new branch. In the majority of cases, you'll always want to create your branches from the `develop` branch.
+
+Before you ever create a new branch, always  checkout the `develop` branch and make sure you you've got the latest changes from `upstream`.
+
+```no-highlight
+$ git checkout develop
+$ git pull upstream develop
+```
+
+!!! warning
+	If you do not do this, you run the risk of having merge conflicts in your branch, and that's never fun to deal with. Trust us on this one.
 
-Typically, you'll base pull requests off of the `develop` branch, or off of `feature` if you're working on a new major release. **Never** merge pull requests into the `main` branch, which receives merged only from the `develop` branch.
+Now that you've got the latest upstream changes, create your branch. It's convention to always prefix your branch name with your GitHub username, separated by hyphens. For example:
+
+```no-highlight
+$ git checkout -b yourusername-myfeature
+```
 
 ## Enabling Pre-Commit Hooks
 
-Nautobot ships with a [git pre-commit hook](https://githooks.com/) script that automatically checks for style compliance and missing database migrations prior to committing changes. This helps avoid erroneous commits that result in CI test failures. You are encouraged to enable it by creating a link to `scripts/git-hooks/pre-commit`:
+Nautobot ships with a [Git pre-commit hook](https://githooks.com/) script that automatically checks for style compliance and missing database migrations prior to committing changes. This helps avoid erroneous commits that result in CI test failures. You are encouraged to enable it by creating a link to `scripts/git-hooks/pre-commit`:
```

Comment:
The pre-commit hook pretty much assumes you're using the virtualenv workflow. We should look into what would be required to make it work with the docker-compose workflow as well, but in the meantime, it might be good to add a disclaimer here.

Reply:
Ah yes, it would be good but if it's assuming those are installed locally, there would need to be another check or setting in the pre-commit-hook script to assert that can be ran from within the container as well.

Reply:
I found a way to get it to detect whether it's in side a virtualenv but would rather tackle this in another PR. Putting the disclaimer in there for now.